### PR TITLE
Metrics Server: use vendored OpenAPI definitions

### LINKED ISSUE #3929: In KEDA Metrics Server use OpenAPI definitions from custom-metrics-apiserver instead.
### Proposal

custom-metrics-apiserver serves OpenAPI spec by default since version [v1.25.0](https://github.com/kubernetes-sigs/custom-metrics-apiserver/releases/tag/v1.25.0) (cf https://github.com/kubernetes-sigs/custom-metrics-apiserver/pull/110).

In KEDA Metrics Server, remove generation of adapter/generated/openapi/zz_generated.openapi.go and use OpenAPI definitions from custom-metrics-apiserver instead.

### Use-Case

_No response_

### Anything else?

_No response_

### PR DESCRIPTION
**Pull request by olivierlemasle**
custom-metrics-apiserver serves OpenAPI spec by default since version [v1.25.0] (cf [PR 110]).

[v1.25.0]: https://github.com/kubernetes-sigs/custom-metrics-apiserver/releases/tag/v1.25.0
[PR 110]: https://github.com/kubernetes-sigs/custom-metrics-apiserver/pull/110

In KEDA Metrics Server, remove generation of `adapter/generated/openapi/zz_generated.openapi.go` and use OpenAPI definitions from custom-metrics-apiserver instead.

### Checklist

- [x] Commits are signed with Developer Certificate of Origin (DCO - [learn more](https://github.com/kedacore/keda/blob/main/CONTRIBUTING.md#developer-certificate-of-origin-signing-your-work))
deployment manifests are modified)*
- [x] Changelog has been updated and is aligned with our [changelog requirements](https://github.com/kedacore/keda/blob/main/CONTRIBUTING.md#Changelog)

Fixes #3929

### CODE REVIEW

**Review by zroubalik — APPROVED** [review]
LGTM, thanks!

File: CHANGELOG.md [hunk 1]
```diff
@@ -90,6 +90,7 @@ To learn more about active deprecations, we recommend checking [GitHub Discussio
 - **General**: Bump Golang to 1.18.6 ([#3205](https://github.com/kedacore/keda/issues/3205))
 - **General**: Bump `github.com/Azure/azure-event-hubs-go/v3` ([#2986](https://github.com/kedacore/keda/issues/2986))
 - **General**: Migrate from `azure-service-bus-go` to `azservicebus` ([#3394](https://github.com/kedacore/keda/issues/3394))
+- **General**: Metrics Server: use OpenAPI definitions served by custom-metrics-apiserver
```
Comment (zroubalik) [inline CHANGELOG.md:93]:
```suggestion
- **General**: Metrics Server: use OpenAPI definitions served by custom-metrics-apiserver ([#3929](https://github.com/kedacore/keda/issues/3929))
```

Comment (olivierlemasle) [reply CHANGELOG.md]: Thanks @zroubalik , I was not sure if an issue was required.

**Pull request merged**